server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactory.java

```diff
@@ -17,6 +17,12 @@
 
 import static org.elasticsearch.search.aggregations.support.AggregationUsageService.OTHER_SUBTYPE;
 
+/**
+ * Aggregator factories are responsible for creating the per-shard aggregator instances.  They should select the aggregator instance type
+ * based on the type of the data being aggregated (using the {@link org.elasticsearch.search.aggregations.support.ValuesSourceRegistry}
+ * when appropriate), and any optimizations that factory can make.  The factory layer is the correct place for heuristics to select
+ * different operating modes (such as using Global Ordinals or not).
+ */
```

Comment:
I typically wrap these manually around 80 or 100 characters to make them easier to read inline. Our layout tools don't force it so you don't have to do it, but it's a thing I like.

Also I'm super ok importing `ValuesSourceRegistry` even though it's only used by the comment. It doesn't hurt anything.